base/array.jl

```diff
@@ -1987,3 +1987,139 @@ julia> symdiff([1,2,3],[3,4,5],[4,5,6])
 ```
 """
 symdiff(a, b, rest...) = symdiff(a, symdiff(b, rest...))
+
+## replace/replace! ##
+
+# NOTE: to implement replace! and replace for a new type T, it's enough to define:
```

Comment:
This note is intended also for people adding a new method in Base... What do you think about keeping/removing this note @tkelman ?